rir/src/compiler/pir/instruction.h

```diff
@@ -94,13 +94,14 @@ class Instruction : public Value {
 
     Instruction(Tag tag, PirType t) : Value(t, tag) {}
 
-    bool mightIO() const;
-    bool changesEnv() const;
-    bool leaksEnv() const;
-    bool hasEnv() const;
-    bool accessesEnv() const;
+    virtual bool mightIO() const = 0;
+    virtual bool changesEnv() const = 0;
+    virtual bool leaksEnv() const = 0;
+    virtual bool hasEnv() const = 0;
+    virtual bool accessesEnv() const = 0;
```

Comment:
the problem was that I could not use the predicates in static assert. so, for example the `static_assert(Description.HasEnv)` was not possible, because I don't think we can create the table as a constexpr.